sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,445 @@
+     *
+     * @param blobUrl URI of the blob.
+     * @param deleteOptions Delete options for the blob and its snapshots.
+     * @param blobAccessConditions Additional access conditions that must be met to allow this operation.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> delete(String blobUrl, DeleteSnapshotsOptionType deleteOptions,
+        BlobAccessConditions blobAccessConditions) {
+        return deleteHelper(getUrlPath(blobUrl), deleteOptions, blobAccessConditions);
+    }
+
+    private Response<Void> deleteHelper(String urlPath, DeleteSnapshotsOptionType deleteOptions,
+        BlobAccessConditions blobAccessConditions) {
+        setBatchType(BlobBatchType.DELETE);
+        return createBatchOperation(batchClient.deleteWithResponse(deleteOptions, blobAccessConditions),
+            urlPath, EXPECTED_DELETE_STATUS_CODES);
+    }
+
+    /**
+     * Adds a set tier operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobBatch.setTier#String-String-AccessTier}
+     *
+     * @param containerName The container of the blob.
+     * @param blobName The name of the blob.
+     * @param accessTier The tier to set on the blob.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> setTier(String containerName, String blobName, AccessTier accessTier) {
+        return setTierHelper(String.format("%s/%s", containerName, blobName), accessTier, null);
+    }
+
+    /**
+     * Adds a set tier operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobBatch.setTier#String-String-AccessTier-LeaseAccessConditions}
+     *
+     * @param containerName The container of the blob.
+     * @param blobName The name of the blob.
+     * @param accessTier The tier to set on the blob.
+     * @param leaseAccessConditions Lease access conditions that must be met to allow this operation.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> setTier(String containerName, String blobName, AccessTier accessTier,
+        LeaseAccessConditions leaseAccessConditions) {
+        return setTierHelper(String.format("%s/%s", containerName, blobName), accessTier, leaseAccessConditions);
```

Comment:
String.format is very expensive in terms of performance. I would not recommend using it in a high-volume code path (maybe okay for exceptions) but it's better to use StringBuilder or just plain string concat (which gets converted to StringBuilder by the compiler internally).

https://kylewbanks.com/blog/java-string-concatenation-vs-stringbuilder-vs-string-format-performance
https://stackoverflow.com/questions/513600/should-i-use-javas-string-format-if-performance-is-important
https://redfin.engineering/java-string-concatenation-which-way-is-best-8f590a7d22a8

Reply:
This is like any other method that does string manipulation, it should only trigger once for each method call and shouldn't normally be called in a tight loop.